app/src/main/java/com/example/othregensburg/zapp/QRCodeGenerator.kt

```diff
@@ -0,0 +1,33 @@
+package com.example.othregensburg.zapp
+
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import android.widget.Button
+import android.widget.EditText
+import android.widget.ImageView
+import com.google.zxing.BarcodeFormat
+import com.journeyapps.barcodescanner.BarcodeEncoder
+
+class QRCodeGenerator : AppCompatActivity() {
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_q_r_code_generator)
+
+        // Get Id´s to work with
+        val etText = findViewById<EditText>(R.id.et_text)
+        val btnGen = findViewById<Button>(R.id.btn_generate)
+        val ivBarcode = findViewById<ImageView>(R.id.iv_barcode)
+
+        // Button clicked -> generates QR Code from input - for test purposes the input is a TextView
+        btnGen.setOnClickListener {
+            try {
+                val encoder = BarcodeEncoder()
+                val bitmap = encoder.encodeBitmap(etText.text.toString(), BarcodeFormat.QR_CODE, 500, 500 ) //First Parameter: what´s hidden in QR Code - AES Encryption
```

Comment:
Using the XZing library is perfectly fine. Why would you add AES Encryption? What is the reason behind?

Reply:
We weren´t sure if we have to implement an encryption for the QR-Code. Do we have to? 
In reality without an encryption a user could technically take a photo of the QR-Code and send it to others for multiple usage - do we have to consider this or would it be enough to give only the barkeepers permission to create one by typing into a TextView?